File: core/src/galenscovell/hinterstar/things/ships/InteriorOverlay.scala
package galenscovell.hinterstar.things.ships

import com.badlogic.gdx.scenes.scene2d.ui.Table
import galenscovell.hinterstar.generation.interior.Tile
import galenscovell.hinterstar.things.ships

import scala.collection.mutable.Map


class InteriorOverlay(ship: Ship) extends Table {
  private val interiorParser: ships.InteriorParser = new ships.InteriorParser(ship)
  private val tiles: Array[Array[Tile]] = interiorParser.getTiles

  private val subsystems: Map[String, Tile] = {
    val subsystemMap: Map[String, Tile] = Map()
    for (row: Array[Tile] <- tiles) {
      for (tile: Tile <- row) {
        if (tile.isSubsystem) {
          subsystemMap.put(tile.getName, tile)
        }
      }
    }
    subsystemMap
  }

  construct()


  private def construct(): Unit = {
//    this.setDebug(true)
    for (row: Array[Tile] <- tiles) {
      for (tile: Tile <- row) {
        this.add(tile).width(interiorParser.tileSize).height(interiorParser.tileSize)
      }
      this.row
    }
  }

  def getSubsystemMap: Map[String, Tile] = {
    subsystems
  }
}

File: core/src/galenscovell/hinterstar/things/ships/InteriorOverlay.kt
package galenscovell.hinterstar.things.ships

import com.badlogic.gdx.scenes.scene2d.ui.Table
import galenscovell.hinterstar.generation.interior.Tile



class InteriorOverlay(ship: Ship) : Table() {
    private val interiorParser: InteriorParser = InteriorParser(ship)
    private val tiles: List<List<Tile>> = setup()
    private val subsystems: MutableMap<String, Tile> = createSubsystemMap()



    private fun setup(): List<List<Tile>> {
        interiorParser.parse()
        val t: List<List<Tile>> = interiorParser.getTiles()

        // this.setDebug(true)
        for (row: List<Tile> in t) {
            for (tile: Tile in row) {
                this.add(tile).width(interiorParser.tileSize.toFloat()).height(interiorParser.tileSize.toFloat())
            }
            this.row()
        }

        return t
    }

    private fun createSubsystemMap(): MutableMap<String, Tile> {
        val subsystemMap: MutableMap<String, Tile> = mutableMapOf()
        for (row: List<Tile> in tiles) {
            for (tile: Tile in row) {
                if (tile.isSubsystem()) {
                    subsystemMap.put(tile.getSubsystemName(), tile)
                }
            }
        }
        return subsystemMap
    }

    fun getSubsystemMap(): MutableMap<String, Tile> {
        return subsystems
    }
}
